test/parallel/test-worker-heap-snapshot.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const { Worker } = require('worker_threads');
+const { once } = require('events');
+
+// Ensure that worker.getHeapSnapshot() returns a valid JSON
+(async () => {
+  const worker = new Worker('setInterval(() => {}, 1000);', { eval: true });
+  await once(worker, 'online');
+  const stream = await worker.getHeapSnapshot();
```

Comment:
Hope this one isn't flaky like the other one

Reply:
@benjamingr @Trott which one was that? I was talking with @ErickWendel today and I might have some idea of what the problem was/is

Reply:
Looks like it was already fixed https://github.com/nodejs/node/issues/41204 ?

Reply:
Hm, that test is actually very very similar to this one. That also means that I’m wrong about having an idea about the cause of the flakiness :)